tqdm.py

```diff
@@ -0,0 +1,42 @@
+import cardinality  # You might need to do 'pip install cardinality'.
+import time
+
+PROGRESS_BAR_LEN = 20
+
+
+class Tqdm(object):
+
+    def __init__(self, iterable):
+        self.iterable = iterable
+        self.length = cardinality.count(iterable)
+
+    def __iter__(self):
+        return TqdmIterator(iter(self.iterable), self.length)
+
+
+class TqdmIterator(object):
+
+    def __init__(self, iterator, length):
+        self.start_time = time.time()
+        self.length = length
+        self.iterator = iterator
+        self.iter_index = 0
+
+    def __next__(self):
+        print('\r' + self._get_progress_bar(), end='')
+        self.iter_index += 1
+        return next(self.iterator)
+
+    def _get_percent(self) -> float:
+            return self.iter_index / self.length * 100
+
+    def _get_progress_bar(self) -> str:
+        percent = self._get_percent()
+        passed = int(PROGRESS_BAR_LEN * (percent/100))
```

Comment:
Since `percent` is used only in this line, there's no need to multiply by 100 and then divide by 100.

Reply:
precent is used at return